backend/data/src/main/kotlin/io/tolgee/MtServicesConfiguration.kt

```diff
@@ -32,18 +32,17 @@ class MtServicesConfiguration(
   }
 
   @Bean
-  fun getAwsTranslationService(): AmazonTranslate? {
-
-    if (awsMachineTranslationProperties.accessKey != null) {
-      return AmazonTranslateClient.builder().withCredentials(
-        AWSStaticCredentialsProvider(
-          BasicAWSCredentials(
-            awsMachineTranslationProperties.accessKey,
-            awsMachineTranslationProperties.secretKey
-          )
-        )
-      ).withRegion(awsMachineTranslationProperties.region).build()
+  fun getAwsTranslationService(): TranslateClient? {
+    val chain = when (awsMachineTranslationProperties.accessKey.isNullOrEmpty()
+                   || awsMachineTranslationProperties.secretKey.isNullOrEmpty()) {
+      true -> DefaultCredentialsProvider.create()
```

Comment:
So how do we authenticate, when we don't provide the credentials? Am I missing something?

Reply:
@JanCizmar if you read the Javadoc here, it's pretty comprehensive, trying all manners of authentication in order. This responds to config files, environment variables, IAM roles, EC2 stuff, etc. all without special effort on our part: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/auth/credentials/DefaultCredentialsProvider.html

Reply:
Oh, I see. I am wondering what is the use case for this. Is it just the simplicity of the configuration or is there also some must have?

Reply:
Oh it's a huge **must have**. We can't actually move forward with self hosting until we have this. It's a violation of many security policies (including ours) to use access key and secret key now and instead you use delegated roles.